Assets/Scripts/Managers/PlacementSystem/PlacementManager.cs

```diff
@@ -150,8 +152,22 @@ private void OnPlaceStructure() {
 		if (_canPlace) {
 			GameObject newPlacement = Instantiate(_buildablePrefab, _currentPlacedPrefab.transform.position, _currentPlacedPrefab.transform.rotation);
 			newPlacement.transform.parent = _placementPosition.transform;
+			newPlacement.AddComponent<MeshCollider>();
+			newPlacement.layer = 9;
 
 			EventBus.Instance.TriggerEvent<GameObject>(EventType.CHANGE_STRUCTURE, _buildablePrefab);
 		}
 	}
+
+	private void OnRemoveStructure() {
+		int layer = 9;
```

Comment:
Magic numbers are bad!

Try and add a layermask serializableObject like the _buildableLayer above.